packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -417,6 +422,34 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 		hostSnapshotOptions: ISnapshotOptions | undefined,
 		scenarioName?: string,
 	) {
+		// Don't look into cache, if the host specifically tells us so.
+		if (!this.hostPolicy.avoidPrefetchSnapshotCache) {
```

Comment:
Is avoidPrefetchSnapshotCache mostly a safety net? I.e. if we have a bug, app can disable this code path?
Or do you think there is legit workflow where host wants to disable this feature?
It feels like it's the former, and if so, it feels like risk is low - they can always work-around an issue by skipping / awaiting prefetch call before launching load (slower, but avoids new code path).

I'm simply looking for simplicity - it's very hard for our customers to understand full matrix of options...

Reply:
No, it is not for that. Cosmin mentioned that he wants to have some hostpolicy in case they want to do a good A/B testing for performance like when they call prefetch api and then load the container, then how fast is to load if we use the prefetch promise from cache and when we are not. So, they can write their logic and use this policy to achieve that.
We don't need a feature gate because if some error occurred while prefetching then, according to current logic I am simply just proceeding with the network call according to old flow.
Once we have more confidence then I can just throw if the prefetch fails and not make a network call.

Reply:
They can just avoid this option and not pass anything if they don't want to use this.

Reply:
I'd actually suggest not doing A/B testing like that. In a sense that we know that leveraging in-flight prefetch would be always faster than doing another parallel call.
So really, we are after only measuring by how much it would be faster. And we can ensure we have enough telemetry in place to measure that, without requiring A/B testing.

The only place where A/B testing would be useful if we think that (for some reason) leveraging this new code would make things worse. But I'd think that's more about reliability than perf, as I have hard time seeing where it would be slower that way. Plus, we have natural A/B testing here in the form of old vs. new driver - we will have at least 1-2 weeks where we have both in usage and can easily do this kind of comparison.

At the end of the day it's up to Loop, but I'd make sure that
- we remove this option in next - cleaner / simpler code
- we do have enough telemetry in place to answer all these questions directly.

Reply:
We've always struggled with assessing the actual gains of perf improvements by looking exclusively at old vs new code sessions (code changes in these layer may be landing at different times in hosts, prefetch may have been called or not on some scenarios, etc). The reliable way to measure the impact of this change itself would be to focus on scenarios telemetry that (1) do prefetch so those sessions are legit for experimentation and (2) split those into A/B. Hence I suggested the extra flags. Seeing the gains accurately would help.

If you think old/new telemetry without the A/B flag is actually allowing for as accurate assessment, due to large volume of sessions maybe and focusing on one single scenario etc, then I'm ok with that. It's just less headache in data digging when you can A/B. But agreed, the code can look gnarly when forcing A/B flags.

Reply:
I am going to leave as is for now. It is just an if statement. We can always remove later on.

Reply:
I agree that overall doing it via version differences is hard. But in this case, I'd argue we should be able to tell precisely (per session) what the impact of these changes is. Maybe other way to say it - we should use all the tools we have here, but I expect A/B testing data will be noisier than direct telemetry (using only new bits) on how much time was saved in a single TreesLatest call by observing the diff between when prefetch started and when Container Load path started to wait for that in-flight call. I want to make sure that we have telemetry to collect those direct measurements.

Reply:
Recording this time now too.